Source/Core/InputCommon/DynamicInputTextures/DITConfiguration.cpp

```diff
@@ -54,197 +53,25 @@ Configuration::Configuration(const std::string& json_file)
     return;
   }
 
-  const picojson::value& output_textures_json = out.get("output_textures");
-  if (!output_textures_json.is<picojson::object>())
-  {
-    ERROR_LOG_FMT(
-        VIDEO,
-        "Failed to load dynamic input json file '{}' because 'output_textures' is missing or "
-        "was not of type object",
-        json_file);
-    m_valid = false;
-    return;
-  }
-
-  const picojson::value& preserve_aspect_ratio_json = out.get("preserve_aspect_ratio");
-
-  bool preserve_aspect_ratio = true;
-  if (preserve_aspect_ratio_json.is<bool>())
-  {
-    preserve_aspect_ratio = preserve_aspect_ratio_json.get<bool>();
-  }
-
-  const picojson::value& generated_folder_name_json = out.get("generated_folder_name");
+  SplitPath(json_file, &m_base_path, nullptr, nullptr);
 
-  const std::string& game_id = SConfig::GetInstance().GetGameID();
-  std::string generated_folder_name = fmt::format("{}_Generated", game_id);
-  if (generated_folder_name_json.is<std::string>())
+  const picojson::value& specification_json = root.get("specification");
+  u8 specification = 1;
+  if (specification_json.is<double>())
   {
-    generated_folder_name = generated_folder_name_json.get<std::string>();
+    specification = static_cast<u8>(specification_json.get<double>());
```

Comment:
This is undefined behavior if the JSON value happens to be non-representable as a `u8`. See https://github.com/dolphin-emu/dolphin/pull/9480 (MathUtil::SaturatingCast) for an example of the clamping you need to do...

Reply:
Good catch.  I was shocked when `picojson` didn't have integer support.

Guess I'll check if it's between the range of 0 and `std::numeric_limts<u8>::max()` and if not mark it invalid.